File: Python/ByRound/1974/1974_A_Phone_Desktop.py
# Problem: CF 1974 A - Phone Desktop
# https://codeforces.com/contest/1974/problem/A

"""
Algorithm: Greedy approach to minimize the number of 5x3 screens needed.
Each screen has 15 cells (5 * 3).

Approach:
1. First, place all 2x2 icons optimally. Since each 2x2 icon occupies 4 cells and a screen has 15 cells,
   we can fit at most 3 such icons per screen, but actually, we can place up to 4 (2x2) icons on a 
   screen if they're arranged in a pattern like a 4x4 square, which doesn't fit in 5x3. 
   So the optimal approach is to pack them as efficiently as possible.

2. However, a better method:
   - Try to fit the 2x2 icons first.
   - For y 2x2 icons, we need at least ceil(y / 2) screens because 
     each screen can hold at most 2 such icons (since 2*2 = 4 and 4 <= 15).
   - Then, calculate how many 1x1 icons remain after placing 2x2 icons.
   - For remaining 1x1 icons, pack them greedily, using full rows or partial rows on the remaining screens.

3. A much simpler greedy idea:
   - Start by calculating how many screens are needed just for 2x2 icons.
   - Place them greedily.
   - Count how many 1x1 icons are leftover after placing 2x2 icons.
   - Allocate extra screens as needed.

But the current algorithm is:
- Calculate a = ceil(y / 2)
- Try to place remaining 1x1 icons after assigning one 2x2 icon per screen.
- Then greedily allocate leftover 1x1 on top of existing or new screens.

Time Complexity: O(1) - only basic arithmetic operations.
Space Complexity: O(1) - only a fixed number of variables used.
"""

import sys

input = lambda: sys.stdin.readline().strip()
printerr = lambda *args, **kwargs: print(
    "\u001B[31m", *args, "\u001B[0m", file=sys.stderr, **kwargs
)


def main():
    x, y = map(int, input().split()) # x = 1x1 apps, y = 2x2 apps
    a = (y + 1) // 2  # Estimate of minimum screens for 2x2 apps (at most 2 per screen)
    x -= 4 * (y % 2)  # Remove space used by 2x2 icons (each requires a fixed amount)  
    x -= a * 7        # Deduct capacity of 1x1 icons for current number of screens, assuming 7 cells left per screen
    a += max((x + 14) // 15, 0)  # Add more screens needed for leftover 1x1 apps
    print(a)


tcs = int(input())
for tc in range(tcs):
    main()


# https://github.com/VaHiX/CodeForces/

File: CSharp/ByRound/1974/1974_A_Phone_Desktop.cs
// Problem: CF 1974 A - Phone Desktop
// https://codeforces.com/contest/1974/problem/A

/*
 * Problem: Phone Desktop
 * 
 * Purpose: Given x 1x1 icons and y 2x2 icons, place them on minimum number of 5x3 screens.
 * 
 * Algorithm:
 * - Each screen has 15 cells.
 * - 2x2 icons take 4 cells each; 1x1 icons take 1 cell each.
 * - Strategy:
 *   1. Place all 2x2 icons first optimally.
 *   2. For each 2x2 icon, we can fill up to 4 cells of a screen.
 *   3. A screen can hold at most 3 (2x2) icons (since 3*4 = 12 cells, leaving 3), or less.
 *   4. So, we try to use full or partially full screens.
 *   5. After placing all 2x2 icons, calculate how many more 1x1 icons fit.
 *   6. Each 1x1 icon takes 1 cell.
 *   7. Each screen holds 15 cells.
 * 
 * Time Complexity: O(1) - constant time per test case.
 * Space Complexity: O(1) - only using a fixed amount of variables.
 */

using System;
public class PhoneDesktop {
  public static void Main() {
    short t = short.Parse(Console.ReadLine());
    for (short i = 0; i < t; i++) {
      string[] parts = Console.ReadLine().Split();
      byte x = byte.Parse(parts[0]);
      byte y = byte.Parse(parts[1]);
      
      // Calculate minimum number of screens needed for 2x2 icons
      // If y is even, we can fit y/2 screens completely (each 2x2 icon takes 4 cells)
      // If y is odd, we need one extra screen for the last 2x2 icon
      byte ans = Convert.ToByte(y % 2 == 0 ? y / 2 : (y + 1) / 2);
      
      // Calculate how many 1x1 icons can be placed in the screen(s) already used for 2x2 icons
      // We start by finding the number of 1x1s we can fit after placing 2x2 icons
      short numSmall = Convert.ToInt16(y % 2 == 0 ? ans * 7 : (ans - 1) * 7 + 11);
      
      // If the number of 1x1 icons exceeds what we can accommodate, add additional screens
      if (numSmall < x) {
        byte d = Convert.ToByte(x - numSmall);
        // Calculate how many additional screens are needed for remaining 1x1 icons
        ans += Convert.ToByte(d % 15 == 0 ? d / 15 : d / 15 + 1);
      }
      
      Console.WriteLine(ans);
    }
  }
}


// https://github.com/VaHiX/CodeForces/